src/agent/plugins/build/prepare.ts

```diff
@@ -49,69 +50,59 @@ export function beforeJob(executionContext: cm.IExecutionContext, callback) {
     });
 
     if (srcendpoints.length == 0) {
-        callback(new Error('No valid repository type'));
-        return;
+        throw new Error('Unsupported SCM system.  Supported: ' + supported.toString());
     }
-
-    // only support 1
+    
+    // only support 1 SCM system
     var endpoint: agentifm.ServiceEndpoint = srcendpoints[0];
 
-    var sys = variables[cm.sysVars.system];
-    var collId = variables[cm.sysVars.collectionId];
-
-    var defId = variables[cm.sysVars.definitionId];
-    var hashInput = collId + ':' + defId;
-
     //
-    // Get the repo path under the working directory
+    // Get SCM plugin
     //
-    var hashInput = collId + ':' + defId;
-    if (job.environment.endpoints) {
-        job.environment.endpoints.forEach(function (endpoint) {
-            hashInput = hashInput + ':' + endpoint.url;
-        });
-    }
-    // TODO: build dir should be defined in the build plugin - not in core agent
-    var hashProvider = crypto.createHash("sha256");
-    hashProvider.update(hashInput, 'utf8');
-    var hash = hashProvider.digest('hex');
-    var workingFolder = variables[cm.agentVars.workingDirectory];
-    var buildDirectory = path.join(workingFolder, sys, hash);
-
-    executionContext.info('using build directory: ' + buildDirectory);
-
-    job.environment.variables['agent.buildDirectory'] = buildDirectory;
-    shell.mkdir('-p', buildDirectory);
-    shell.cd(buildDirectory);
-
-    var repoPath = path.resolve('repo');
-    job.environment.variables['build.sourceDirectory'] = repoPath;
-    job.environment.variables['build.stagingdirectory'] = path.resolve("staging");
-
-    // TODO: remove compat variable
-    job.environment.variables['sys.sourcesFolder'] = repoPath;
-
     var scmm;
     var providerType = endpoint.type.toLowerCase();
     executionContext.info('using source provider: ' + providerType);
 
-    try {
+    try {        
         var provPath = path.join(executionContext.scmPath, providerType);
         executionContext.info('loading: ' + provPath);
-        scmm = require(provPath);    
+        scmm = require(provPath);
     }
     catch(err) {
         callback(new Error('Source Provider not found: ' + providerType));
```

Comment:
I changed the error message to could not load provider.
I did the opposite.  I changed the other two to callbacks.  We shouldn't be throwing since this is still a callback convention in prepare plugin